blockdag/dag.go

```diff
@@ -1058,21 +1058,24 @@ func (dag *BlockDAG) meldVirtualUTXO(newVirtualUTXODiffSet *DiffUTXOSet) error {
 	return newVirtualUTXODiffSet.meldToBase()
 }
 
-func (node *blockNode) diffFromTxs(pastUTXO UTXOSet, transactions []*util.Tx) (*UTXODiff, error) {
-	diff := NewUTXODiff()
+func (node *blockNode) calcUTXO(pastUTXO *DiffUTXOSet, transactions []*util.Tx) (UTXOSet, error) {
```

Comment:
`CalcUTXO` is a bit too general IMHO. I'd rename to `applyTransactionsToUTXO` or something